slapo/op/mlp.py

```diff
@@ -23,12 +27,14 @@ def __init__(self, hidden_size, intermediate_size, orig_act, resid_pdrop):
         else:
             raise NotImplementedError(f"Unsupported activation: {orig_act}")
 
-        self.fc_out = torch.nn.Linear(intermediate_size, hidden_size)
-        self.dropout = torch.nn.Dropout(resid_pdrop)
+        self.fc_out = torch.nn.Linear(intermediate_size, hidden_size, bias=False)
+        self.bias_dropout = FusedBiasDropout(
+            hidden_size, resid_pdrop, prev_weight=self.fc_out.weight
+        )
```

Comment:
This is actually a good point and I also realized that in these days. I may try to refactor our ops to avoid this case as possible.

Reply:
I just found that it doesn't matter in FusedMLP, because when replacing an entire MLP, you will need to consider parameter conversion anyways, because even the name `fc_out` may not match the name in the original MLP. Assuming the original MLP has `fc_1` and `fc_2`, then we have
```
mlp.fc1.weight -> mlp.fc_in.weight
mlp.fc1.bias -> mlp.act.bias
mlp.fc2.weight
```

Reply:
Parameter initialization is not based on the attribute name but is based on the module class. See the HF [`_init_weights`](https://github.com/huggingface/transformers/blob/main/src/transformers/models/bert/modeling_bert.py#L746-L760). This method is applied recursively, which means it does not need to a mapping convertion.

Reply:
I see so you're talking about pre-training. I was referring to model checkpoints. That's a fair call then.

Reply:
Okay, for common cases, I would expect module replacement does not change the original behavior. If the parameter values are also changed, it will be hard to verify or future debug.